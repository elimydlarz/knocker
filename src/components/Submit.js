import React, { Component } from 'react';
import '../css/Submit.css';
import transmit from '../services/transmit.js';

export default class TextInput extends Component {
  constructor(props) {
    super(props);
    this.transmit = this.transmit.bind(this);
  }

  transmit(event) {
    event.preventDefault();

    const auth = {
      user: this.props.data.knocker,
      password: this.props.password
    }

    transmit(auth, 'knocked', this.props.data, response => {
      const responseHandler = response.status === 201
        ? this.props.successFn
        : this.props.failureFn;

      responseHandler(response);
    });
  }

  valid() {
    return this.props.requiredFields.every(
      field =>
        field && field.length > 0
    )
  }

  render() { return (
    <div className='Submit'>
      <input type="submit"
        disabled={!this.valid()}
        onClick={this.transmit} />
    </div>
  ); }
}
