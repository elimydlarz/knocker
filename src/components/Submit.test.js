import React from 'react';
import renderer from 'react-test-renderer';
import Submit from './Submit';

describe('Submit', () => {
  it('does not explode', () => {
    const component = renderer.create(
      <Submit
        user='COOL USER'
        password='COOL PASSWORD'
        eventType='COOL EVENT TYPE'
        data={{ coolKey: 'COOL VALUE' }}
        requiredFields={[
          'COOL REQUIRED FIELD'
        ]}
        successFn={() => {}}
        failureFn={() => {}}
      />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
