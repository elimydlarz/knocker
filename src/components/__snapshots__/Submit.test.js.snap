exports[`Submit does not explode 1`] = `
<div
  className="Submit">
  <input
    disabled={false}
    onClick={[Function bound transmit]}
    type="submit" />
</div>
`;
